expr/functions/consolidateBy/function_test.go

```diff
@@ -0,0 +1,83 @@
+package consolidateBy
+
+import (
+	"github.com/go-graphite/carbonapi/expr/helper"
+	"github.com/go-graphite/carbonapi/expr/metadata"
+	"github.com/go-graphite/carbonapi/expr/types"
+	"github.com/go-graphite/carbonapi/pkg/parser"
+	th "github.com/go-graphite/carbonapi/tests"
+	"testing"
+	"time"
+)
+
+func init() {
+	md := New("")
+	evaluator := th.EvaluatorFromFunc(md[0].F)
+	metadata.SetEvaluator(evaluator)
+	helper.SetEvaluator(evaluator)
+	for _, m := range md {
+		metadata.RegisterFunction(m.Name, m.F)
+	}
+}
+
+func TestConsolidateBy(t *testing.T) {
+	now32 := time.Now().Unix()
+
+	tests := []th.EvalTestItem{
+		{
+			"consolidateBy(metric1,\"sum\")",
+			map[parser.MetricRequest][]*types.MetricData{
+				{"metric1", 0, 1}: {types.MakeMetricData("metric1", []float64{1, 2, 3, 4, 5}, 1, now32)},
+			},
+			[]*types.MetricData{types.MakeMetricData("consolidateBy(metric1,\"sum\")",
+				[]float64{1, 2, 3, 4, 5}, 1, now32).SetTag("consolidateBy", "sum").SetConsolidationFunc("sum")},
```

Comment:
I think we would need to add the `actual.ConsolidateFunc != want.ConsolidateFunc` check in this helper for the assertion to be made

https://github.com/go-graphite/carbonapi/blob/50b983a0ac3d230c88067b151b553c92a2be41ef/tests/helper.go#L399